Hot reload : Failed to start quarkus, Already a default codec registered for class class java.util.UUID

**Describe the bug**
(Describe the problem clearly and concisely.)
Whenever the hot reload is triggered, it fails to restart the app after upgrade to Quarkus 1.6.1
**Expected behavior**
The application is reloaded when the code is updated
**Actual behavior**
```
java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.dev.appstate.ApplicationStateNotification.waitForApplicationStart(ApplicationStateNotification.java:51)
	at io.quarkus.runner.bootstrap.StartupActionImpl.runMainClass(StartupActionImpl.java:156)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.restartApp(IsolatedDevModeMain.java:139)
	at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.doScan(RuntimeUpdatesProcessor.java:173)
	at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$1.handle(VertxHttpHotReplacementSetup.java:55)
	at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$1.handle(VertxHttpHotReplacementSetup.java:45)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:313)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:715)
	at io.quarkus.runtime.Application.start(Application.java:90)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:91)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:61)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:30)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at io.quarkus.runner.bootstrap.StartupActionImpl$3.run(StartupActionImpl.java:145)
	... 1 more
Caused by: java.lang.IllegalStateException: Already a default codec registered for class class java.util.UUID
	at io.vertx.core.eventbus.impl.CodecManager.registerDefaultCodec(CodecManager.java:132)
	at io.vertx.core.eventbus.impl.EventBusImpl.registerDefaultCodec(EventBusImpl.java:195)
	at io.quarkus.vertx.runtime.VertxRecorder.registerCodecs(VertxRecorder.java:153)
	at io.quarkus.vertx.runtime.VertxRecorder.configureVertx(VertxRecorder.java:40)
	at io.quarkus.deployment.steps.VertxProcessor$build-626852132.deploy_0(VertxProcessor$build-626852132.zig:183)
	at io.quarkus.deployment.steps.VertxProcessor$build-626852132.deploy(VertxProcessor$build-626852132.zig:36)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:558)
	... 11 more
```

--- a/extensions/vertx/deployment/src/test/java/io/quarkus/vertx/devmode/ConsumeUuidEventHotReloadTest.java
+++ b/extensions/vertx/deployment/src/test/java/io/quarkus/vertx/devmode/ConsumeUuidEventHotReloadTest.java
@@ -1,0 +1,57 @@
+package io.quarkus.vertx.devmode;
+
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusDevModeTest;
+import io.restassured.RestAssured;
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.Router;
+
+public class ConsumeUuidEventHotReloadTest {
+
+    @RegisterExtension
+    static final QuarkusDevModeTest test = new QuarkusDevModeTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(UuidMessageConsumer.class, BeanDeployingAVerticleFromInstance.class));
+
+    private static final String SAMPLE_UUID = "38400000-8cf0-11bd-b23e-10b96e4ef00d";
+
+    @Test
+    public void testUuidMessageConsumption() {
+        String resp = RestAssured.get("/").asString();
+        Assertions.assertEquals("test-" + SAMPLE_UUID, resp);
+        test.modifySourceFile(UuidMessageConsumer.class, data -> data.replace("test-", "other-"));
+        resp = RestAssured.get("/").asString();
+        Assertions.assertEquals("other-" + SAMPLE_UUID, resp);
+        String resp2 = RestAssured.get("/").asString();
+        Assertions.assertEquals(resp, resp2);
+
+    }
+
+    @ApplicationScoped
+    public static class BeanDeployingAVerticleFromInstance {
+        @Inject
+        Vertx vertx;
+
+        public void init(@Observes Router router) throws InterruptedException {
+            CountDownLatch latch = new CountDownLatch(1);
+            vertx.deployVerticle(new MyVerticle(),
+                    ar -> latch.countDown());
+            router.get("/").handler(rc -> vertx.eventBus().<String> request("event", UUID.fromString(SAMPLE_UUID),
+                    ar -> rc.response().end(ar.result().body())));
+            latch.await();
+        }
+    }
+
+}

--- a/extensions/vertx/deployment/src/test/java/io/quarkus/vertx/devmode/UuidMessageConsumer.java
+++ b/extensions/vertx/deployment/src/test/java/io/quarkus/vertx/devmode/UuidMessageConsumer.java
@@ -1,0 +1,16 @@
+package io.quarkus.vertx.devmode;
+
+import java.util.UUID;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import io.quarkus.vertx.ConsumeEvent;
+
+@ApplicationScoped
+public class UuidMessageConsumer {
+
+    @ConsumeEvent(value = "event", blocking = true)
+    public String handleUuid(UUID uuid) {
+        return "test-" + uuid.toString();
+    }
+}

--- a/extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/VertxRecorder.java
+++ b/extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/VertxRecorder.java
@@ -14,6 +14,7 @@ import io.quarkus.runtime.LaunchMode;
 import io.quarkus.runtime.RuntimeValue;
 import io.quarkus.runtime.ShutdownContext;
 import io.quarkus.runtime.annotations.Recorder;
+import io.quarkus.runtime.configuration.ProfileManager;
 import io.quarkus.vertx.ConsumeEvent;
 import io.vertx.core.AsyncResult;
 import io.vertx.core.Handler;
@@ -144,12 +145,19 @@ public class VertxRecorder {
     @SuppressWarnings("unchecked")
     private void registerCodecs(Map<Class<?>, Class<?>> codecByClass) {
         EventBus eventBus = vertx.eventBus();
+        boolean isDevMode = ProfileManager.getLaunchMode() == LaunchMode.DEVELOPMENT;
         for (Map.Entry<Class<?>, Class<?>> codecEntry : codecByClass.entrySet()) {
             Class<?> target = codecEntry.getKey();
             Class<?> codec = codecEntry.getValue();
             try {
                 if (MessageCodec.class.isAssignableFrom(codec)) {
                     MessageCodec messageCodec = (MessageCodec) codec.newInstance();
+                    if (isDevMode) {
+                        // we need to unregister the codecs because in dev mode vert.x is not reloaded
+                        // which means that if we don't unregister, we get an exception mentioning that the
+                        // codec has already been registered
+                        eventBus.unregisterDefaultCodec(target);
+                    }
                     eventBus.registerDefaultCodec(target, messageCodec);
                 } else {
                     LOGGER.error(String.format("The codec %s does not inherit from MessageCodec ", target.toString()));

